src/apps/vaporwave-finance/aurora/vaporwave-finance.farm.contract-position-fetcher.ts

```diff
@@ -0,0 +1,101 @@
+import { Inject } from '@nestjs/common';
+import { compact } from "lodash";
+import Axios from 'axios';
+import { buildDollarDisplayItem } from "~app-toolkit/helpers/presentation/display-item.present";
+
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { Register } from '~app-toolkit/decorators';
+import { PositionFetcher } from '~position/position-fetcher.interface';
+import { ContractPosition } from '~position/position.interface';
+import { Network } from '~types/network.interface';
+
+import { VaporwaveFinanceContractFactory } from '../contracts';
+import { VAPORWAVE_FINANCE_DEFINITION } from '../vaporwave-finance.definition';
+import { getBaseERC20Token } from './vaporwave-finance.vault.token-fetcher';
+import { claimable, supplied } from "~position/position.utils";
+import { ContractType } from "~position/contract.interface";
+import auroraStakePools from './aurora_stake'
+
+
+const appId = VAPORWAVE_FINANCE_DEFINITION.id;
+const groupId = VAPORWAVE_FINANCE_DEFINITION.groups.farm.id;
+const network = Network.AURORA_MAINNET;
+
+@Register.ContractPositionFetcher({ appId, groupId, network })
+export class AuroraVaporwaveFinanceFarmContractPositionFetcher implements PositionFetcher<ContractPosition> {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(VaporwaveFinanceContractFactory)
+    private readonly vaporwaveFinanceContractFactory: VaporwaveFinanceContractFactory,
+  ) { }
+
+  async getPositions() {
+    // http://localhost:5001/apps/vaporwave-finance/positions?groupIds[]=farm&network=aurora
+    const wantPrices = await Axios.get("https://api.vaporwave.farm/lps").then(
+      (v) => v.data
+    )
+    const baseTokenPrices = await Axios.get("https://api.vaporwave.farm/prices").then(
+      (v) => v.data
+    )
```

Comment:
You should use `@CacheOnInterval` to reduce calls to your API, see https://studio.zapper.fi/docs/recipes/caching-patterns for details.